src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -150,12 +176,19 @@ private IntPtr GetPin(Object obj)
             }
             return handle.AddrOfPinnedObject();
         }
-        private void FlushPins()
+
+        private List<IntPtr> _nativeExceptions = new List<IntPtr>();
+
+        private void CompileMethodCleanup()
         {
             foreach (var pin in _pins)
                 pin.Value.Free();
             _pins.Clear();
 
+            foreach (IntPtr ex in _nativeExceptions)
+                FreeException(ex);
+            _nativeExceptions.Clear();
```

Comment:
`nativeExceptions = null` would be better instead of Clear - once you start creating the list lazily.